x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel_wrapper.tsx

```diff
@@ -4,25 +4,82 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
-import { EuiPageContent, EuiPageContentHeader, EuiPageContentBody } from '@elastic/eui';
+import React, { useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import classNames from 'classnames';
+import {
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiPageContentHeader,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+import { FramePublicAPI, Visualization } from '../../types';
+import { NativeRenderer } from '../../native_renderer';
+import { Action } from './state_management';
 
-interface Props {
-  title: string;
+export interface WorkspacePanelWrapperProps {
   children: React.ReactNode | React.ReactNode[];
+  framePublicAPI: FramePublicAPI;
+  visualizationState: unknown;
+  activeVisualization?: Visualization;
+  dispatch: (action: Action) => void;
+  title?: string;
 }
 
-export function WorkspacePanelWrapper({ children, title }: Props) {
+export function WorkspacePanelWrapper({
+  children,
+  framePublicAPI,
+  visualizationState,
+  activeVisualization,
+  dispatch,
+  title,
+}: WorkspacePanelWrapperProps) {
+  const setVisualizationState = useCallback(
+    (newState: unknown) => {
+      if (!activeVisualization) {
+        return;
+      }
+      dispatch({
+        type: 'UPDATE_VISUALIZATION_STATE',
+        visualizationId: activeVisualization.id,
+        newState,
+        clearStagedPreview: false,
+      });
+    },
+    [dispatch]
+  );
   return (
-    <EuiPageContent className="lnsWorkspacePanelWrapper">
-      {title && (
-        <EuiPageContentHeader className="lnsWorkspacePanelWrapper__pageContentHeader">
-          <span data-test-subj="lns_ChartTitle">{title}</span>
-        </EuiPageContentHeader>
+    <EuiFlexGroup gutterSize="s" direction="column" alignItems="stretch">
+      {activeVisualization && activeVisualization.renderToolbar && (
+        <EuiFlexItem grow={false}>
+          <NativeRenderer
+            render={activeVisualization.renderToolbar}
+            nativeProps={{
+              frame: framePublicAPI,
+              state: visualizationState,
+              setState: setVisualizationState,
+            }}
+          />
+        </EuiFlexItem>
       )}
-      <EuiPageContentBody className="lnsWorkspacePanelWrapper__pageContentBody">
-        {children}
-      </EuiPageContentBody>
-    </EuiPageContent>
+      <EuiFlexItem>
+        <EuiPageContent className="lnsWorkspacePanelWrapper">
+          <EuiPageContentHeader
+            className={classNames('lnsWorkspacePanelWrapper__pageContentHeader', {
+              'lnsWorkspacePanelWrapper__pageContentHeader--unsaved': !title,
+            })}
+          >
+            <span data-test-subj="lns_ChartTitle">
+              {title ||
+                i18n.translate('xpack.lens.chartTitle.unsaved', { defaultMessage: 'Unsaved' })}
+            </span>
+          </EuiPageContentHeader>
```

Comment:
Can we make sure the content header only shows when there's a chart? So don't show "Unsaved" in the empty state (alongside the graphic).

Reply:
How important is this to you? It would require to refactor how the components are set up because currently, the workspace panel wrapper doesn't know anything about the rendered chart.

Reply:
@flash1293 It is very important and pretty similar to it used to behave. Before, it would show the title component once you have a visualization and then saved it. Now I'm just asking if we can put a second layer of logic to show the "Unsaved" title after chart creation.

Reply:
@cchaos I withdrew my comment, I found a nice solution :) Sorry for the noise.